### PR TITLE
feat: [STO-6606]: Mount volume with shared certificates for STO

### PR DESCRIPTION
**Pull request by r0610205**
## Describe your changes

STO Plugin 1.19.0 successfully recognizes the certificates coming from the delegate:
image

Build Pod configuration (has HARNESS_SHARED_CERTS_PATH env set and directory populated with certificates coming from delegate):
image

Delegate is running with the following update in the configuration to mount test.pem (created as a configmap)
```
        volumeMounts:
        - name: certs
          mountPath: /opt/harness-delegate/ca-bundle/ca.pem
          subPath: test.pem
      volumes:
        - name: certs
          configMap:
            name: cert-file
            items:
              - key: test.pem
                path: test.pem
```

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54270)

### CODE REVIEW

**Review by williamwissemann — APPROVED** [review]

**Review by GabeAboy — APPROVED** [review]

**Comment by r0610205** [comment]
trigger messagecheck

**Comment by r0610205** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 1]
```diff
@@ -293,6 +300,20 @@ private void updatePodWithDelegateVolumes(CoreV1Api coreV1Api, String namespace,
       return;
     }
 
+    // Create volume and volume mount for shared certificates folder
+    String sharedCertsSourcePath = SystemWrapper.getenv(SHARED_CA_CERTS_PATH);
+    V1HostPathVolumeSource sharedCertsVolumeSource =
+        new V1HostPathVolumeSourceBuilder().withPath(sharedCertsSourcePath).build();
+    podVolumes.add(new V1VolumeBuilder()
+                       .withHostPath(sharedCertsVolumeSource)
```
Comment (xingchi-jin) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:308]: i am not expert on how hostPath mount works in k8s.
the directory resides in delegate container, the LE is another pod. will hostPath mount work ?

**Review by williamwissemann — APPROVED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java (hunk 1 above)
Comment (r0610205) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java]: you are correct, I needed to use the same approach as used in other place. Updated the PR

**Comment by r0610205** [comment]
trigger uts

**Review by xingchi-jin — APPROVED** [review]

**Pull request merged**